src/Type/IntegerType.php

```diff
@@ -139,6 +140,18 @@ public function isScalar(): TrinaryLogic
 
 	public function looseCompare(Type $type, PhpVersion $phpVersion): BooleanType
```

Comment:
I think we have a new todo for PHPStan 3.x: adjust all `Type` methods which depend on `PhpVersion` and turn them into using `PhpVersions`

Reply:
Yeah, this should be possible. It's possible to make PhpVersions from PhpVersion (without losing precision), but not vice-versa.

So places that only have PhpVersion could still call these methods by transforming it into PhpVersions first.